### PR TITLE
Add build and test jobs to CI

### PR DESCRIPTION
**Pull request by JackyWoo**
### change log
1. add build and test jobs to CI

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+jobs:
+  build-and-test:
+    name: "ubuntu clang"
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
```
[the remark this answers is not in the document]
Comment (JackyWoo) [reply .github/workflows/build-and-test.yml:47]: Yes we should pull all codes

File: tests/integration/helpers/cluster_service.py [hunk 2]
```diff
@@ -377,19 +363,19 @@ def start(self, destroy_dirs=True):
...
 
-            start_deadline = time.time() + 180.0  # seconds
             for instance in self.instances.values():
                 instance.docker_client = self.docker_client
                 instance.ip_address = self.get_instance_ip(instance.name)
 
                 print("Waiting for RaftKeeper-server start...")
-                instance.wait_for_start(start_deadline)
+                # instance.wait_for_start(timeout=180)
+                instance.wait_for_join_cluster()
```
[the remark this answers is not in the document]
Comment (JackyWoo) [reply tests/integration/helpers/cluster_service.py:372]: Before `wait_for_join_cluster` we invoke start all instances.  Here we intend to wait RaftKeeper cluster initialized.

File: tests/integration/test_auth/test.py [hunk 3]
```diff
@@ -380,7 +361,7 @@ def test_auth_snapshot(started_cluster):
         connection.create(f"/test_snapshot_acl/path{i}", b"data", acl=[make_acl("auth", "", all=True)])
 
     node.restart_raftkeeper(kill=True)
-    wait_node(started_cluster)
+    node.wait_for_join_cluster()
```
[the remark this answers is not in the document]
Comment (JackyWoo) [reply tests/integration/test_auth/test.py:364]: `restart_raftkeeper`  is for one node.  `wait_for_join_cluster ` is for cluster.

File: tests/integration/test_converter/test.py [hunk 4]
```diff
@@ -8,40 +8,53 @@
 
 cluster = RaftKeeperCluster(__file__)
 
-node = cluster.add_instance('node', main_configs=['configs/keeper_config.xml', 'configs/logs_conf.xml'], stay_alive=True)
+node = cluster.add_instance('node', main_configs=['configs/keeper_config.xml', 'configs/logs_conf.xml'],
+                            stay_alive=True)
```
[the remark this answers is not in the document]
Comment (JackyWoo) [reply tests/integration/test_converter/test.py:12]: Ignore it, I will fix whole integration test code sytle.

File: tests/integration/helpers/cluster_service.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicelulu) [reply tests/integration/helpers/cluster_service.py:372]: ```wait_for_join_cluster()``` use the ZK client connect to node, need cluster quorum. But at this time may not reach the quorum.

File: tests/integration/test_auth/test.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (nicelulu) [reply tests/integration/test_auth/test.py:364]: Can use ```start_wait=True```

File: tests/integration/helpers/cluster_service.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (JackyWoo) [reply tests/integration/helpers/cluster_service.py:372]: Here we wait cluster initialized.

**Pull request merged**